document new stream page action sidebar [AC-4296] (#588)
canonical/anbox-cloud-docs: reference/stream-page-actions.md

Author: Addressed

Reviewer: We need to add this to the ToC.

Reviewer 2: This, and also the PR description mentions:

> Changes are in "How to guides", "Manage applications", "Stream Page Applications" *(I guess this should be "Actions")*

But actually this is nested under `reference`. Should we move it where the PR description says?

Reviewer: Ah, Stefan actually moved it to reference because I thought it didn't really belong in how tos

Reviewer 2: Ah, okay, no problem then. Maybe we should keep the description aligned though, to avoid further misunderstandings :)

# Stream Page Actions

When you are streaming an application or a device, a sidebar on the right side of the stream page provides access to several controls and information panels.

![Stream Page Actions Sidebar](/images/stream-sidebar/streaming-actions-sidebar-page.png)

This sidebar contains the following actions:

## Developer tools

![Developer Tools Toggle](/images/stream-sidebar/developer-tools-toggle-icon.png)

*Shortcut: `Ctrl + Shift + D`*

The **Developer tools** action opens a panel with various tools for developers to interact with the streaming session, including a terminal and logs for the running instance.

## Share

![Share Stream](/images/stream-sidebar/share-icon.png)

*Shortcut: `Ctrl + Shift + S`*

> The **Share** action creates a temporary link to share your current streaming session with others. This is useful for collaboration or for demonstrating an application to someone who does not have an account.

Reviewer 2: ```suggestion
The **Share** action creates a temporary link to share your stream with others. This is useful for collaboration or for demonstrating an application to someone who does not have an account.
```
Maybe we should warn users that simultaneous access is disallowed? I think the second sentence might trick readers into thinking that they could share the stream and join it simultaneously from different devices - which is not the case.

## Connect ADB

![Connect ADB](/images/stream-sidebar/connect-adb-icon.png)

*Shortcut: `Ctrl + Shift + C`*

> The **Connect ADB** action provides [instructions](/howto/android/access-android-instance.md) to connect to the Android Debug Bridge (ADB) for the running instance. This allows developers to interact with the Android instance using ADB commands, which is useful for debugging and testing applications.

Reviewer 2: It's not just instructions - you actually create the pre-signed URL from here. Can we slightly reword this to reflect the actual feature of this button?

## Statistics

![Statistics Toggle](/images/stream-sidebar/statistics-toggle-icon.png)

*Shortcut: `Ctrl + Shift + M`*

The **Statistics** action opens a panel that displays real-time streaming statistics. This information is useful for monitoring the performance and quality of the stream and can help diagnose streaming issues.

## Full screen

![Full Screen Toggle](/images/stream-sidebar/full-screen-toggle-icon.png)

*Shortcut: `Ctrl + Shift + F`*

The **Full screen** action toggles the streaming session between full-screen and windowed mode.

> ## Stream resize

Reviewer 2: Should we use the label that we show in the tooltip? For example, this would be "Enable stream resize".

<img width="370" height="120" alt="Image" src="https://github.com/user-attachments/assets/7d4ee26e-0a1b-4e29-bc0a-e51506c38d9a" />

Reviewer 2: Naturally, the same question applies to the other headers in the list, and also to the text highlighted in bold in the descriptions.

Author: Hmm, I think we can

Reviewer 2: > and also to the text highlighted in bold in the descriptions

Looks like this part was left unchanged. Currently we have a mismatch between titles and action highlighted in bold - e.g.:

<img width="406" height="352" alt="image" src="https://github.com/user-attachments/assets/c1004be8-7926-42c8-8c45-68161ee10b68" />

Reviewer 2: Note that the above is just an example. Currently all items have this mismatch, so all need to be fixed. I'm not leaving individual comments on each of them.

![Stream Resize Toggle](/images/stream-sidebar/stream-resize-toggle-icon.png)

*Shortcut: `Ctrl + Shift + R`*

> The **Stream Resize** action allows you to resize the stream.

Reviewer 2: ```suggestion
The **Stream resize** action allows you to toggle resize mode, which allows to resize the stream, or reset its size to fit the whole available space.
```

## Rotate left

![Rotate Left Toggle](/images/stream-sidebar/rotate-left-toggle-icon.png)

*Shortcut: `Ctrl + Shift + ArrowLeft`*

> The **Rotate Left** action rotates the stream to the left by 90 degrees.

Reviewer 2: ```suggestion
The **Rotate left** action rotates the stream to the left by 90 degrees.
```

## Rotate right

![Rotate Right Toggle](/images/stream-sidebar/rotate-right-toggle-icon.png)

*Shortcut: `Ctrl + Shift + ArrowRight`*

> The **Rotate Right** action rotates the stream to the right by 90 degrees.

Reviewer 2: ```suggestion
The **Rotate right** action rotates the stream to the right by 90 degrees.
```

## Volume up

![Volume Up Toggle](/images/stream-sidebar/volume-up-toggle-icon.png)

*Shortcut: `Ctrl + Shift + ArrowUp`*

> The **Volume Up** action increases the volume of the Android instance, simulating a volume button press on a physical device.

Reviewer 2: ```suggestion
The **Volume up** action increases the volume of the Android device, simulating a volume button press on a physical device.
```

## Volume down

![Volume Down Toggle](/images/stream-sidebar/volume-down-toggle-icon.png)

*Shortcut: `Ctrl + Shift + ArrowDown`*

> The **Volume Down** action decreases the volume of the Android instance, simulating a volume button press on a physical device.

Reviewer 2: ```suggestion
The **Volume down** action decreases the volume of the Android device, simulating a volume button press on a physical device.
```

## Power

![Power](/images/stream-sidebar/power-toggle-icon.png)

*Shortcut: `Ctrl + Shift + P`*

> The **Power** action turns the Android instance on or off, simulating a power button press on a physical device.

Reviewer 2: ```suggestion
The **Power** action turns the display of the Android device on or off, simulating a power button press on a physical device.
```

## Location

![Location Toggle](/images/stream-sidebar/location-toggle-icon.png)

The **Location** action opens a panel where you can set the geographic location (GPS coordinates) for the virtual device. This is useful for testing applications that rely on location services.

When you set a location, the virtual device behaves as if it is physically present at those coordinates.

> This action might be missing from the sidebar for AAOS applications. In this case, the location toggle is in the "Controls" panel.

Reviewer 2: This section should be moved above, after the "Statistics" item. Also, it's currently missing the shortcut line.

Reviewer 2: This was moved *before* statistics instead of *after* as suggested in the comment above. Check the order on the page and make sure we match it:

<img width="74" height="232" alt="image" src="https://github.com/user-attachments/assets/9108dcff-9857-41b1-b200-6ddb75df4d6c" />

Reviewer 2: @Stefan3002 see the comment above please. This is still not addressed.

Author: Addressed, my bad

## Bug Report

![Bug Report](/images/stream-sidebar/bug-report-icon.png)

*Shortcut: `Ctrl + Shift + B`*

> The **Bug Report** action provides a way to [capture and submit](/howto/troubleshoot/index.md#bug-report-diagnostic-facility) information about a problem you encounter during a streaming session. When you initiate a bug report, the system collects relevant logs and session data to help developers diagnose the issue.

Reviewer 2: ```suggestion
The **Bug report** action provides a way to [collect](/howto/troubleshoot/index.md#bug-report-diagnostic-facility) information about a problem you encounter during a streaming session. When you initiate a bug report, the system collects relevant logs and session data that you can then send to developers to help them diagnose the issue.
```

> ## Release keyboard

Reviewer 2: ```suggestion
## Toggle keyboard capture
```

![Release Keyboard Toggle](/images/stream-sidebar/release-keyboard-toggle-icon.png)

*Shortcut: `Ctrl + Shift + K`*

> The **Release keyboard** action releases the keyboard focus from the streaming session, allowing you to use your keyboard for other tasks.

Reviewer 2: This is a toggle button, we should specify that it allows to toggle the capture of key events in the Android device on and off.

## Screenshot

![Screenshot Toggle](/images/stream-sidebar/screenshot-toggle-icon.png)

*Shortcut: `Ctrl + Shift + I`*

> The **Screenshot** action takes a screenshot of the Android Instance. The screenshot is taken at the client level, capturing exactly what is displayed on the stream.

Reviewer 2: Do we really need the "client level" bit? Is that something interesting/important for the user?

```suggestion
The **Screenshot** action takes a screenshot of the Android device.
```

## Screen record

![Screen Record Toggle](/images/stream-sidebar/screen-record-toggle-icon.png)

*Shortcut: `Ctrl + Shift + O`*

> The **Screen Record** action records a video of the streaming session. Clicking the action again stops the recording and saves the video to your local device. This is useful for creating demos, tutorials, or for troubleshooting.

Reviewer 2: ```suggestion
The **Screen record** action starts recording a video of the streaming session. Clicking the button a second time stops the recording and saves the video to your local device. This is useful for creating demos, tutorials, or for troubleshooting.
```

(ref-stream-page-actions)=

> # Stream Page Actions

Reviewer 2: Noticed this from the TOC - casing in the title looks a bit inconsistent with the other items:

```suggestion
# Stream page actions
```

When you are streaming an application or a device, a sidebar on the right side of the stream page provides access to several controls and information panels.

![Stream Page Actions Sidebar](/images/stream-sidebar/streaming-actions-sidebar-page.png)

This sidebar contains the following actions:

## Open developer tools

![Developer Tools Toggle](/images/stream-sidebar/developer-tools-toggle-icon.png)

*Shortcut: `Ctrl + Shift + D`*

The **Developer tools** action opens a panel with various tools for developers to interact with the streaming session, including a terminal and logs for the running instance.

## Set up sharing

![Share Stream](/images/stream-sidebar/share-icon.png)

*Shortcut: `Ctrl + Shift + S`*

> The **Share** action creates a temporary link to share your stream with another person, allowing them to take over the session from you. Simultaneous access is not allowed. When the other person joins the stream, you are disconnected.

Reviewer 2: ```suggestion
The **Share** action creates a temporary link to share your stream with another person, allowing them to take over the session from you. Note that simultaneous access is not allowed: when someone joins the stream, the currently connected user is disconnected.
```

## Connect ADB

![Connect ADB](/images/stream-sidebar/connect-adb-icon.png)

*Shortcut: `Ctrl + Shift + C`*

The **Connect ADB** action generates a pre-signed URL that can be used to connect to the Android Debug Bridge (ADB) for the running instance. It also provides [instructions](/howto/android/access-android-instance.md) on how to use the URL. This allows developers to interact with the Android instance using ADB commands, which is useful for debugging and testing applications.

## Set location

*Shortcut: `Ctrl + Shift + G`*

![Location Toggle](/images/stream-sidebar/location-toggle-icon.png)

The **Location** action opens a panel where you can set the geographic location (GPS coordinates) for the virtual device. This is useful for testing applications that rely on location services.

When you set a location, the virtual device behaves as if it is physically present at those coordinates.

> This action might be missing from the sidebar for AAOS applications. In this case, the location toggle is in the "Controls" panel.

Reviewer 2: Actually, it's not a "toggle":

```suggestion
This action might be missing from the sidebar for AAOS applications. In this case, the location section can be opened from the "Controls" panel.
```

## Statistics

![Statistics Toggle](/images/stream-sidebar/statistics-toggle-icon.png)

*Shortcut: `Ctrl + Shift + M`*

> The **Statistics** action opens a panel that displays real-time streaming statistics. This information is useful for monitoring the performance and quality of the stream and can help diagnose streaming issues.

Reviewer 2: ```suggestion
## Show statistics

![Statistics Toggle](/images/stream-sidebar/statistics-toggle-icon.png)

*Shortcut: `Ctrl + Shift + M`*

The **Show statistics** action opens a panel that displays real-time streaming statistics. This information is useful for monitoring the performance and quality of the stream and can help diagnose streaming issues.
```

## Full screen

![Full Screen Toggle](/images/stream-sidebar/full-screen-toggle-icon.png)

*Shortcut: `Ctrl + Shift + F`*

The **Full screen** action toggles the streaming session between full-screen and windowed mode.

## Enable/Disabled stream resize

![Stream Resize Toggle](/images/stream-sidebar/stream-resize-toggle-icon.png)

*Shortcut: `Ctrl + Shift + R`*

> The **Stream resize** action allows you to toggle resize mode, which allows to resize the stream, or reset its size to fit the whole available space.

Reviewer 2: ```suggestion
## Enable / Disable stream resize

![Stream Resize Toggle](/images/stream-sidebar/stream-resize-toggle-icon.png)

*Shortcut: `Ctrl + Shift + R`*

The **Enable / Disable stream resize** action allows you to toggle resize mode, which allows to resize the stream, or reset its size to fit the whole available space.
```

## Volume up

![Volume Up Toggle](/images/stream-sidebar/volume-up-toggle-icon.png)

*Shortcut: `Ctrl + Shift + ArrowUp`*

The **Volume up** action increases the volume of the Android device, simulating a volume button press on a physical device.

## Volume down

![Volume Down Toggle](/images/stream-sidebar/volume-down-toggle-icon.png)

*Shortcut: `Ctrl + Shift + ArrowDown`*

The **Volume down** action decreases the volume of the Android device, simulating a volume button press on a physical device.

## Power

![Power](/images/stream-sidebar/power-toggle-icon.png)

*Shortcut: `Ctrl + Shift + P`*

The **Power** action turns the display of the Android device on or off, simulating a power button press on a physical device.

## Home

![Home Button](/images/stream-sidebar/home-icon.png)

*Shortcut: `Ctrl + Shift + H`*

The **Home** action simulates a home button press on a physical device.

## Back

![Back Button](/images/stream-sidebar/back-icon.png)

*Shortcut: `Ctrl + Shift + E`*

The **Back** action simulates a back button press on a physical device.

## Download bug report

![Bug Report](/images/stream-sidebar/bug-report-icon.png)

*Shortcut: `Ctrl + Shift + B`*

> The **Download bug report** action provides a way to [collect](/howto/troubleshoot/index.md#bug-report-diagnostic-facility) information about a problem you encounter during a streaming session. When you initiate a bug report, the system collects relevant logs and session data that you can then send to developers to help them diagnose the issue.

Reviewer: This is the warning you see in the check. This should point to sec-bug-report using the myst way.
```suggestion
The **Download bug report** action provides a way to {ref}`collect <sec-bug-report>`) information about a problem you encounter during a streaming session. When you initiate a bug report, the system collects relevant logs and session data that you can then send to developers to help them diagnose the issue.
```

(untested, but that should work, according to: https://myst-parser.readthedocs.io/en/latest/syntax/cross-referencing.html#reference-roles

Author: Thanks for the help @ajanon , that was it!

## Release / Capture keyboard

![Release Keyboard Toggle](/images/stream-sidebar/release-keyboard-toggle-icon.png)

*Shortcut: `Ctrl + Shift + K`*

> The **Release / Capture keyboard** action allows you to switch the capture of keyboard events by the Android device on and off. When the capture is off, you can use your keyboard for other tasks on your computer.

Reviewer 2: ```suggestion
The **Release / Capture keyboard** action allows you to switch the capture of keyboard events by the Android device on and off. When the capture is off, the key events will be sent to the browser's page.
```
